### ISSUE TITLE
Empty for-range loop on a channel flagged by empty-block

### ISSUE THREAD

**Issue by feldgendler** [mgechev/revive#861]
**Describe the bug**
An empty `for ... range` loop on a channel gets flagged by `empty-block`.

Unlike empty `for ... range` loops on slices and maps, one on a channel has an effect even when the body is empty: it drains the channel.

**To Reproduce**
`revive@latest`, no configuration file, all defaults, reports this code:

```go
func drain(ch <-chan int) {
    for range ch {}
}
```

**Expected behavior**
`empty-block` is not triggered.

**Logs**
```
test.go:2:2: this block is empty, you can remove it
```

**Desktop:**
 - OS: Ubuntu 23.04
- `go version go1.20.6 linux/amd64`

**Comment by chavacava** [comment]
duplicated of #386

**Comment by feldgendler** [comment]
I can see that this is a duplicate of #386, whose fix was later reverted by #416. Do those reasons still stand? That was in 2020, and some things might have changed since (for instance, Go modules are now the norm).

I can see that many other revive checks use type information. Why can't this one?
